### ISSUE TITLE
Attack and damage roll wrappers do not pass along dialogOptions

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#689]
In GitLab by @kaelad on Feb 8, 2022, 20:16

The `itemhandling.js` file has some wrappers for attack and damage rolls: `doAttackRoll` and `doDamageRoll` respectively. Neither of them pass along `dialogOptions` that are sent as options to those methods. So if any modules or macros that are calling `item.rollAttack` or `item.rollDamage` that pass along some `dialogOptions`, they are ignored. It's possible this could be coming from the Midi's workflow part of the code, I'm not 100% sure though, but I suspect it's in these wrappers.

This is causing an issue with another module. I tried working around it by changing priorities in libWrapper's configuration but it won't work. Midi's methods are "MIXED" so they are always called after "WRAPPED" (which is what they are for the other module).

### Test case using macro

```javascript
const item = actor.items.getName("Greataxe");
await item.rollAttack({
  dialogOptions: { width: 600 }
});
```

1. Change the item name to what you have on a test actor.
2. Make sure that fast-forwarding attack rolls is not enabled so you can actually see the roll dialog.
3. Clear the chat log, just to be safe. I ran into some non-reproducible issue where it actually worked, but couldn't reproduce after clearing the chat log.

**Expected:** The roll dialog should be much wider than normal: 600px instead of 400px.

**Comment by tposney** [comment]
Thanks for the detailed report. I've got a couple of questions.

Are you doing this as part of a workflow (i.e. normal item.roll()) or a standalone attack roll. If the latter whatever options are passed to the item.rollAttack are passed through to the underlying roll.

I tried your example and got the expected wider dialog box. Can you also export your midi settings so I can make sure I'm testing the same environment?

**Comment by tposney** [comment]
In any event I'll ensure that they are passed through, but I'd like to be able to repro the problem.

**Comment by tposney** [comment]
In GitLab by @kaelad on Feb 9, 2022, 24:12

I was doing some compatibility testing with my module, Advantage Reminder, and Midi and ran into the issue during the normal workflow. I'd roll a weapon and Midi would auto-roll the attack. I don't have "Auto fast forward attack" enabled so I expect to see my module change something on the roll dialog that pops up but the change was not there. My module passes something along through `dialogOptions` to change the roll dialog and from what I could tell Midi was not passing `dialogOptions` along or using it.

The macro example was my attempt to give you a reproducible test case without needing to install my module. I thought that would be easier for you. You could give my module a shot though if the macro test case isn't working. You'll need to add an active effect on the test actor with `flags.adv-reminder.message.all | Custom | Message on all rolls` and it should add that message right above the Advantage/Normal/Critical buttons on all roll dialogs. It works on other rolls like ability checks/saves, but not on attack and damage rolls.

Here's my Midi settings I was using to test with. [fvtt-midi-qol-settings-689.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/1915fab9529fb6b95e90f25fe1437c3f1fa01dcfbfc3380b2d44be923e55b5dc/fvtt-midi-qol-settings-689.json)

**Comment by tposney** [comment]
In GitLab by @kaelad on Feb 9, 2022, 24:14

I think I tested with "Auto roll attack" off then manually clicking the Attack button and had the same result

**Comment by tposney** [comment]
I'm not sure what was happening, but I could not repro the test case. However, I installed your module and can see the problem. 

I now pass through all options passed to me (for attacks and damages) overwriting those that I care about and it seems to be playing nicely.

Thanks for the investigation you did, much appreciated and makes fixing much easier (you're right about the cause of the issue).

Fix will go out in 0.9.11
